.circleci/config.yml

```diff
@@ -0,0 +1,40 @@
+version: 2
+jobs:
+   test:
+    docker:
+      - image: ghcr.io/cirruslabs/flutter:3.19.0
+      
+    steps:
+      - checkout
+      - run: flutter doctor
+      - run: flutter pub get
+      - run: flutter pub run build_runner build --delete-conflicting-outputs
+      - run: flutter test
+      - run: flutter analyze .
+      - run: flutter pub publish --dry-run
```

Comment:
Let's remove the publish step from the test job as this can cause undesirable behavior like the package getting released every time we make a change in a PR 😅

Reply:
the `--dry-run` option tells the command to simulate the publishing process without actually publishing the package.

Reply:
Oh sorry just realized we're passing `--dry-run` sorry for the confusion 😅